html/browsers/browsing-the-web/navigating-across-documents/javascript-url-return-value-handling-dynamic.html

```diff
@@ -0,0 +1,48 @@
+<!doctype html>
+<meta charset=UTF-8>
+<title>Test javascript URL string return values in direct and indirect (target) frame contexts.</title>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<div id=log></div>
+<script>
+const testInputs = [
+  [0x41],
+  [0x80,0xFF],
+  [0x80,0xFF,0x100],
+  [0xD83D,0xDE0D],
+  [0xDE0D,0x41]
+];
+testInputs.forEach(input => {
+  const javascriptURL = "javascript:[" + input + "].map(b => String.fromCharCode(b)).join('')",
+        output = input.map(b => String.fromCharCode(b)).join("");
+
+  async_test(t => {
+    const frame = document.createElement("iframe");
+    t.add_cleanup(() => frame.remove());
+    frame.src = javascriptURL;
+    t.step_timeout(() => {
+      assert_equals(frame.contentDocument.body.textContent, output);
+      assert_equals(frame.contentDocument.charset, document.charset);
+      t.done();
+    }, 200);
+    document.body.appendChild(frame);
+  });
+
+  async_test(t => {
+    const frame = document.createElement("iframe"),
+          href = document.createElement("a");
+    t.add_cleanup(() => { frame.remove(); href.remove(); });
+    frame.name = "hi" + input;
+    href.target = "hi" + input;
+    href.href = javascriptURL;
+    t.step_timeout(() => {
+      assert_equals(frame.contentDocument.body.textContent, output);
+      assert_equals(frame.contentDocument.charset, document.charset);
+      t.done();
+    }, 200)
+    document.body.appendChild(frame);
+    document.body.appendChild(href);
+    href.click();
```

Comment:
Wouldn't this ruin the entire harness in user agents that don't use the `target` attribute?

Reply:
I was not ware Firefox did not support it. Anyway latest update should be better as it splits up the frame scenario and the link target scenario.

Reply:
Firefox does, but Chrome/Safari don't, right?

Reply:
Yes, that is correct, and I have a fix ready for WebKit/Safari.
I noticed an error in my change, should be fixed now.